topics/galaxy-ui/tutorials/workflow-parameters/tutorial.md

```diff
@@ -0,0 +1,160 @@
+> 
+> * You can connect a parameter to multiple steps or to mulitple parameters within a step
+{: .tip}
+
+# Create a text parameter from multiple parameters
+
+It is often necessary to create a text parameter that is flanked by additional text.
+Take for instance a regular expression that finds `foo` in a string and replaces it
+with `foobar` as in `s/(foo)/\1bar/`.
+The Galaxy tool ``Regex Find And Replace`` can be used to tun such a regular expression. The find part of the regex (`s/(foo)/`) can be defined in the `Find regex` parameter and the replacement part can be entered in the `Replacement` parameter.
+If we want to make the `foo` part of the regular expression configurable we can
+compose this text parameter using the `Compose text parameter value` tool.
+
+## Compose a text parameter
+1. **Add a data input to the workflow**
+2. **Add a `Simple inputs used for workflow logic` to the workflow**
+   - {% icon param-select %} *"Parameter type"*: `Text`
+3. **Add the `Compose text parameter value` to the workflow** {% icon tool %}
+    - Add three repeats
+    - In the first repeat:
+      - {% icon param-text %} *"Enter text that should be part of the computed value"*: `/(`
+    - In the second repeat:
+      - {% icon param-text %} *"Enter text that should be part of the computed value"*: Leave empty and click "Add connection to module".
+    - In the third repeat:
+      - {% icon param-text %} *"Enter text that should be part of the computed value"*: `)/`.
+    - Connect the `Simple inputs used for workflow logic` to the `Compose text parameter value` parameter input
+4. **Add the `Regex Find And Replace` to the workflow**
+    - Click on `Insert Check`
+    - Click on `Add connection to module` for the `Find Regex` parameter
+    - *"Replacement"*: `\1bar`
+    - Connect the output of the  `Compose text parameter value` tool to the `Find Regex` parameter.
+    - Connect the data input to the `Select lines from` input of the `Regex Find And Replace` tool
+
+Now upload a text dataset with the contents `wunder`
+If you run this workflow on the dataset and you select `wunder` as the newly defined parameter  you will see 2 new datasets in your history. The first dataset has the data type ``expression.json`` and contains the composed parameter value `(wunder)`, the second dataset will contain the output of the `Regex Find And Replace` step. A click on the `i` button will show the used parameters. You will see that the `Find Regex` parameter will contain the values that you entered in the workflow run form. If you look at the dataset content you will see it is `wunderbar`.
+
+# Read a parameter from a dataset
+
+Often times it is necessary to calculate a parameter in one step of a workflow and then to use it in another step of the same workflow. This can be accomplished by reading the parameter from a dataset (As long as it is a text, integert, float, boolean or color parameter).
```

Comment:
```suggestion
Often times it is necessary to calculate a parameter in one step of a workflow and then to use it in another step of the same workflow. This can be accomplished by reading the parameter from a dataset (As long as it is a text, integer, float, boolean or color parameter).
```